rioxarray/rioxarray.py

```diff
@@ -612,72 +630,75 @@ def estimate_utm_crs(self, datum_name: str = "WGS 84") -> rasterio.crs.CRS:
+            if transform is None and zarr.has_convention_declared(
+                self._obj.attrs, "spatial:"
+            ):
+                transform = zarr.read_transform(self._obj)
+            return transform
 
-        except KeyError:
-            try:
-                return Affine(*self._obj.attrs["transform"][:6])
-            except KeyError:
-                pass
         return None
 
     def write_transform(
         self,
         transform: Optional[Affine] = None,
         grid_mapping_name: Optional[str] = None,
+        convention: Optional[Convention] = None,
         inplace: bool = False,
     ) -> Union[xarray.Dataset, xarray.DataArray]:
         """
         .. versionadded:: 0.0.30
 
-        Write the GeoTransform to the dataset where GDAL can read it in.
-
-        https://gdal.org/drivers/raster/netcdf.html#georeference
+        Write the transform to the dataset using the specified convention.
 
         Parameters
         ----------
         transform: affine.Affine, optional
             The transform of the dataset. If not provided, it will be calculated.
         grid_mapping_name: str, optional
             Name of the grid_mapping coordinate to store the transform information in.
-            Default is the grid_mapping name of the dataset.
+            Only used with CF convention. Default is the grid_mapping name of the dataset.
+        convention: Convention, optional
+            Convention to use for writing transform. If None, uses the global default
+            from set_options().
         inplace: bool, optional
             If True, it will write to the existing dataset. Default is False.
 
         Returns
         -------
         :obj:`xarray.Dataset` | :obj:`xarray.DataArray`:
-            Modified dataset with Geo Transform written.
+            Modified dataset with transform written.
         """
         transform = transform or self.transform(recalc=True)
         data_obj = self._get_obj(inplace=inplace)
-        # delete the old attribute to prevent confusion
-        data_obj.attrs.pop("transform", None)
-        grid_mapping_name = (
-            self.grid_mapping if grid_mapping_name is None else grid_mapping_name
-        )
-        try:
-            grid_map_attrs = data_obj.coords[grid_mapping_name].attrs.copy()
-        except KeyError:
-            data_obj.coords[grid_mapping_name] = xarray.Variable((), 0)
-            grid_map_attrs = data_obj.coords[grid_mapping_name].attrs.copy()
-        grid_map_attrs["GeoTransform"] = " ".join(
-            [str(item) for item in transform.to_gdal()]
-        )
-        data_obj.coords[grid_mapping_name].rio.set_attrs(grid_map_attrs, inplace=True)
-        return data_obj.rio.write_grid_mapping(
-            grid_mapping_name=grid_mapping_name, inplace=True
-        )
+
+        # Determine which convention to use
+        if convention is None:
+            convention = get_option(CONVENTION) or Convention.CF
```

Comment:
Here is a another instance retrieving the convention taking into account user input.